reporter/iface.go

```diff
@@ -54,7 +55,13 @@ type SymbolReporter interface {
 
 	// ExecutableMetadata accepts a fileID with the corresponding filename
 	// and caches this information before a periodic reporting to the backend.
-	ExecutableMetadata(ctx context.Context, fileID libpf.FileID, fileName, buildID string)
+	//
+	// The `open` argument can be used to open the executable for reading. Interpreters
+	// that don't support this may pass a `nil` function pointer. Implementations that
+	// wish to upload executables should NOT block this function to do so and instead just
+	// open the file and then enqueue the upload in the background.
+	ExecutableMetadata(ctx context.Context, fileID libpf.FileID, fileName, buildID string,
```

Comment:
With a type for wrapping all the arguments, we make future extensions less intrusive. I can think of additional future information that may make sense (e.g. the full pathname).